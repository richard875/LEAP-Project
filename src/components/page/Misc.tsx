"use client";

import Image from "next/image";
import { useTheme } from "next-themes";
import { useEffect, useState } from "react";
import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/avatar";

const Misc = () => {
  const { systemTheme } = useTheme();
  const [mounted, setMounted] = useState(false);

  useEffect(() => {
    setMounted(true);
  }, []);

  if (!mounted) return null;

  return (
    <>
      {systemTheme === "light" ? (
        <Image
          className="fixed top-6 left-5 select-none"
          src="/logo-light.webp"
          alt="LawConnect Logo"
          width={172.2}
          height={23.8}
          priority
        />
      ) : (
        <Image
          className="fixed top-6 left-5 select-none"
          src="/logo-dark.webp"
          alt="LawConnect Logo"
          width={172.2}
          height={23.8}
          priority
        />
      )}

      <Avatar className="fixed top-5 right-5 select-none">
        <AvatarImage src="https://github.com/richard875.png" />
        <AvatarFallback>RL</AvatarFallback>
      </Avatar>
    </>
  );
};

export default Misc;
