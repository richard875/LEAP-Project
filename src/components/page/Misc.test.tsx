import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import Misc from "./Misc";

describe("Misc", () => {
  it("should render correctly", () => {
    const { container } = render(<Misc />);
    expect(container).toBeTruthy();
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(<Misc />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct elements in the component", () => {
    render(<Misc />);
    const avatar = screen.getByText("RL");
    const logoLight = screen.getByAltText("LawConnect Logo");

    expect(avatar).toBeInTheDocument();
    expect(logoLight).toBeInTheDocument();
  });
});
