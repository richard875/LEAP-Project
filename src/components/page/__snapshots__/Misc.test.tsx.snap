// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Misc should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-12 w-full fixed top-0 left-0 border-b bg-background block xl:hidden"
  />
  <img
    alt="LawConnect Logo"
    class="fixed top-3 left-0 select-none xl:top-6 xl:left-5 scale-75 xl:scale-100"
    data-nimg="1"
    decoding="async"
    height="23.8"
    src="/_next/image?url=%2Flogo-dark.webp&w=384&q=75"
    srcset="/_next/image?url=%2Flogo-dark.webp&w=256&q=75 1x, /_next/image?url=%2Flogo-dark.webp&w=384&q=75 2x"
    style="color: transparent;"
    width="172.2"
  />
  <span
    class="size-8 shrink-0 overflow-hidden rounded-full fixed top-5 right-5 select-none hidden xl:flex"
    data-slot="avatar"
  >
    <span
      class="bg-muted flex size-full items-center justify-center rounded-full"
      data-slot="avatar-fallback"
    >
      RL
    </span>
  </span>
</DocumentFragment>
`;
